contrib/guix/libexec/build.sh

```diff
@@ -0,0 +1,410 @@
+
+# Build the depends tree, overriding variables that assume multilib gcc
+make -C contrib/depends --jobs="$JOBS" HOST="$HOST" \
+                                   ${V:+V=1} \
+                                   ${SOURCES_PATH+SOURCES_PATH="$SOURCES_PATH"} \
+                                   ${BASE_CACHE+BASE_CACHE="$BASE_CACHE"} \
+                                   ${SDK_PATH+SDK_PATH="$SDK_PATH"} \
+                                   OUTDIR="$OUTDIR" \
+                                   LOGDIR="$LOGDIR" \
+                                   x86_64_linux_CC=x86_64-linux-gnu-gcc \
+                                   x86_64_linux_CXX=x86_64-linux-gnu-g++ \
+                                   x86_64_linux_AR=x86_64-linux-gnu-gcc-ar \
+                                   x86_64_linux_RANLIB=x86_64-linux-gnu-gcc-ranlib \
+                                   x86_64_linux_NM=x86_64-linux-gnu-gcc-nm \
+                                   x86_64_linux_STRIP=x86_64-linux-gnu-strip
+
+# Log the depends package hashes
+DEPENDS_PACKAGES="$(make -C contrib/depends --no-print-directory HOST="$HOST" print-all_packages)"
+DEPENDS_CACHE="$(make -C contrib/depends --no-print-directory ${BASE_CACHE+BASE_CACHE="$BASE_CACHE"} print-BASE_CACHE)"
+
+# Keep a record of the depends packages and their hashes that will be used for
+# our build. If there is a reproducibility issue, comparing this log file could
+# help narrow down which package is responsible for the defect.
+{
+    for package in ${DEPENDS_PACKAGES}; do
+        cat "${DEPENDS_CACHE}/${HOST}/${package}"/*.hash
+    done
+} | sort -k2 > "${LOGDIR}/depends-packages.txt"
+
+# Stop here if we're only building depends packages. This is useful when
+# debugging reproducibility issues in depends packages. Skips ahead to the next
+# target, so we don't spend time building Monero binaries.
+if [[ -n "$DEPENDS_ONLY" ]]; then
+    exit 0
+fi
+
+###########################
+# Source Tarball Building #
+###########################
+
+# Use COMMIT_TIMESTAMP for the source and release binary archives
+export SOURCE_DATE_EPOCH=${COMMIT_TIMESTAMP}
+export TAR_OPTIONS="--owner=0 --group=0 --numeric-owner --mtime='@${SOURCE_DATE_EPOCH}' --sort=name"
+
+GIT_ARCHIVE="${DIST_ARCHIVE_BASE}/monero-source-${VERSION}.tar.gz"
+
+# Create the source tarball if not already there
+# This uses `git ls-files --recurse-submodules` instead of `git archive` to make
+# sure submodules are included in the source archive.
+if [ ! -e "$GIT_ARCHIVE" ]; then
+    mkdir -p "$(dirname "$GIT_ARCHIVE")"
+    git ls-files --recurse-submodules \
+    | cat \
+    | sort \
+    | tar --create --transform "s,^,monero-source-${VERSION}/," --mode='u+rw,go+r-w,a+X' --files-from=- \
+    | gzip -9n > ${GIT_ARCHIVE}
+    sha256sum "$GIT_ARCHIVE"
+fi
+
+###########################
+# Binary Tarball Building #
+###########################
+
+# CFLAGS
+HOST_CFLAGS="-O2"
```

Comment:
I wonder if this would be clearer if you just had a `*` in the case statement and pushed these two lines into the case, where appropriate. Is the `find` command even helpful on non-linux/gnu hosts?

Reply:
I cleaned this up. We have logic to set the optimization level for Release builds inside `CMakeLists.txt`, so setting it here is redundant/unnecessary.